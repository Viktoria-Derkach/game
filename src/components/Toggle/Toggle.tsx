import React, { FC } from 'react';
import styled from '@emotion/styled';

export interface ToggleProps {
  children: string;
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export const Toggle: FC<ToggleProps> = React.memo(({ children, onClick }) => (
  <Button onClick={onClick}>{children}</Button>
));

Toggle.displayName = 'Toggle';

const Button = styled.button`
  margin: 0 0.3rem;
  padding: 0.3rem 0.7rem;
  color: #e6e6e6;
  background-color: #383b40;
  border-radius: 0.5rem;
  border: 0.5px solid #010101;
  font-family: 'Roboto', sans-serif;
  cursor: pointer;
  &:hover {
    background-color: #383b405e;
  }
`;
