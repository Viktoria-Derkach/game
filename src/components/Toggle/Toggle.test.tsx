import React, { FC } from 'react';
import { render } from '@testing-library/react';

import { Toggle } from './Toggle';

describe('Toggle button check', () => {
  it('Toggle render check', () => {
    const { asFragment } = render(
      <Toggle onClick={jest.fn()}>Stop/Start</Toggle>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
