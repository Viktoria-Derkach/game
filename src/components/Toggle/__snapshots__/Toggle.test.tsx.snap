// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle button check Toggle render check 1`] = `
<DocumentFragment>
  <button
    class="css-2ly8m7"
  >
    Stop/Start
  </button>
</DocumentFragment>
`;
